### PR TITLE
Refactor sf:auth to use cli.action

### PR DESCRIPTION
**Pull request by gulopine**
Connects to heroku/herokuconnect#2612

### CODE REVIEW

File: lib/commands/connect/sf-auth.js [hunk 1]
```diff
@@ -31,24 +38,30 @@ module.exports = {
...
-    console.log("Launching Salesforce for authorization. If your browser doesn't open, please copy the following URL to proceed:\n\n" + redir + "\n");
-    yield cli.open(redir);
+    let redir;
+
+    yield cli.action('fetching authorizing URL', co(function* () {
+      let connection = yield api.withConnection(context, heroku);
+
+      let url = '/api/v3/connections/' + connection.id + '/authorize_url';
+      let args = {
+        'environment':'production',
```
Comment (sibson) [inline lib/commands/connect/sf-auth.js]: should be a flag?

**Comment by sibson** [comment]
:thumbsup: :shipit:

File: lib/commands/connect/sf-auth.js (hunk 1 above)
Comment (gulopine) [inline lib/commands/connect/sf-auth.js]: I had wondered about that as well. We can either add all the flags now or push this out and increment later. I had been leaning toward the latter, but I could be persuaded that it's worth the effort now.

Comment (sibson) [inline lib/commands/connect/sf-auth.js]: I don't want to think about custom domains, so willing to punt for now

**Pull request merged**